connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/CatalystDataToProtobuf.scala

```diff
@@ -26,7 +26,8 @@ import org.apache.spark.sql.types.{BinaryType, DataType}
 private[protobuf] case class CatalystDataToProtobuf(
     child: Expression,
     messageName: String,
-    descFilePath: Option[String] = None)
+    descFilePath: Option[String] = None,
+    options: Map[String, String] = Map.empty)
```

Comment:
@rangadi I don't see options being used anywhere in the process of converting CatalystData to Protobuf. do we need it?

Reply:
We will likely add more options in the future. It is currently used in Databricks distribution for couple of things:
  - Enable support for sharded java classes (Databricks disables it by default due to complexity of shading)
  - Schema registry related options. 
I can imagine future options like supporting json serialization (rather than binary bytes).

Reply:
Agree. Options like 'JSON format' will be in open source (not working on it yet).
I think only schema-registry support is the only one missing. It is a bit tricky to get it in.